File: lib/domain/models/animal_category.dart
class AnimalCategory {
  final int id;
  final int speciesId;
  final String name;
  final DateTime? archivedAt;

  const AnimalCategory({
    required this.id,
    required this.speciesId,
    required this.name,
    this.archivedAt,
  });

  bool get isArchived => archivedAt != null;
}

File: src/domain/models/animal-category.ts
import { z } from 'zod';

export const AnimalCategory = z.object({
  id: z.string(),
  speciesId: z.string(),
  label: z.string(),
  ordering: z.number().int(),
  isCustom: z.boolean(),
  archivedAt: z.string().nullable(),
});

export type AnimalCategory = z.infer<typeof AnimalCategory>;
